p2p/protocols/accounting_test.go

```diff
@@ -168,16 +173,25 @@ func TestBalance(t *testing.T) {
 }
 
 func checkAccountingTestCases(t *testing.T, cases []testCase, acc *Accounting, peer *Peer, balance *dummyBalance, send bool) {
+	t.Helper()
 	for _, c := range cases {
 		var err error
-		var expectedResult int64
+		var expectedResult, cost int64
 		//reset balance before every check
 		balance.amount = 0
 		if send {
-			err = acc.Send(peer, c.size, c.msg)
+			cost, err = acc.Validate(peer, c.size, c.msg, Sender)
+			if err != nil {
+				t.Fatal(err)
+			}
+			err = acc.Apply(peer, cost, c.size)
 			expectedResult = c.sendResult
```

Comment:
c.sendResult is always 0 (see comment above). I think it would be good to add an additional test case here which would catch this scenario.

Reply:
expectedResult here comes from the `costToLocalNode` (result of accounting.Validate), but later (`p2p/protocols/accounting_test.go:209`), this variable is compared to the balance. This does not seem correct to me. Anyways, tests pass incorrectly as all is just zero.

Reply:
Are you sure about this?

Try printing `result`, it is not always zero.

Reply:
I put myself on a wrong track of thought because I misinterpreted c.msg to be a non-priced message, while they are actually priced. This comment can be ignored.